Feature/polygon api light service update

philip-morlier opened the pull request.

philip-morlier committed:
Initial work to update polygon plugin to be consistent with flume light and heavy.

Moved get_chainId from the polygon plugin into the main project api.

philip-morlier committed:
marged changes from origin develop

philip-morlier committed:
Eth api extended to implement light/heavy configuration.

philip-morlier committed:
Bor namespace extended to implement light/heavy configuration

AusIV commented on plugins/packages/polygon/main.go:
```diff
@@ -100,3 +99,47 @@ func getBlockAuthor(header *evm.Header) (common.Address, error) {
+	case types.Hash:
+		blockHash := input.(types.Hash)
+		var response int
+		statement := "SELECT 1 FROM bor.bor_logs WHERE blockHash = ?;"
```
I think we should bease this on blocks.blocks rather t han bor.bor_logs. Bor logs are only present on 1/64 blocks, so this would mean 63/64 blocks would get routed to Flume heavy even if Flume light has those blocks.

AusIV commented on plugins/packages/polygon/main.go:
```diff
@@ -100,3 +99,47 @@ func getBlockAuthor(header *evm.Header) (common.Address, error) {
+	return present
+}
+
+func borTxDataPresent(txHash types.Hash, cfg *config.Config, db *sql.DB) bool {
```
Is this function even used?

AusIV commented on plugins/packages/polygon/main.go:
```diff
@@ -100,3 +99,47 @@ func getBlockAuthor(header *evm.Header) (common.Address, error) {
+		present = false
+		return present
+	}
+	mpStatement := "SELECT 1 FROM mempool.transactions WHERE hash = ?;"
```
Bor transactions will never be on the mempool table, so I don't think we need to check here for bor specific things.

philip-morlier committed:
Refinement of extention of polygon eth and bor apis to implement light/heavy configuration.

Alteration of block methods to return nil instead of empty maps.

Wrapping inputs to block method calls to heavy server.

philip-morlier committed:
getSnapshot() modified to accept hashes and numbers as well as light/configuration.

philip-morlier committed:
Elminate extraneous utility functions.

philip-morlier committed:
Improved error handling in getSnapshot().

philip-morlier committed:
Further improvements to error handling getSnapshot.

AusIV commented on plugins/packages/polygon/snapshot.go:
```diff
@@ -284,7 +351,7 @@ func (service *PolygonBorService) GetSnapshot(ctx context.Context, blockNrOrHash
 			if err != nil {
 				log.Error("GetSnapshot keyframe case error")
-				return nil, err
+				return nil, nil
```
Is returning a nil error in this case correct behavior?

philip-morlier replied:
My feeling is that the public method should not return errors as it could potentially expose our database structure to users. I went through and standardized error handling in this file. Private methods return errors as well as logs them, whereas the public method only logs them.

philip-morlier committed:
Updated imports in plugin to reflect cardinal-flume name change.

philip-morlier committed:
Merged origin/develop and resolive conflicts

philip-morlier committed:
Updated error handling in snapshot.go.
